fss/src/app/rest.service.ts

```diff
@@ -84,9 +84,15 @@ export class RestService {
     }
 
     public getCurrentUserId(): Promise<number> {
-        let url = `${this.getBaseUrl()}/user?username=${this.settings.current_username}`;
+        let url = `${this.getBaseUrl()}/user`;
         return this.http.get(url, { headers: this.getHeaders() })
-        .toPromise().then(response => { console.log(response); return response[0]["id"]; }).catch(this.handleError);
+        .toPromise().then(response => { 
+            console.log(response); 
+            let users = response as User[];
+            console.log(users);
```

Comment:
Maybe remove the logging statements before merging? Other than that it looks good.